gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultKeystoreService.java

```diff
@@ -300,6 +310,27 @@ public Key getKeyForGateway(String alias, char[] passphrase) throws KeystoreServ
     }
   }
 
+  @Override
+  public Key getKeyForGateway(char[] passphrase) throws KeystoreServiceException {
+    return getKeyForGateway(config.getIdentityKeyAlias(), passphrase);
+  }
+
+  @Override
+  public Certificate getCertificateForGateway() throws KeystoreServiceException, KeyStoreException {
+    Certificate cert = null;
+    readLock.lock();
```

Comment:
Just a minor note: `getKeystoreForGateway` already applies read lock; do we need to lock here? I mean the only critical step here is to fetch the `ks` instance which is already guarded by the `readLock`.

Reply:
Agree.. nice catch. Fixing.